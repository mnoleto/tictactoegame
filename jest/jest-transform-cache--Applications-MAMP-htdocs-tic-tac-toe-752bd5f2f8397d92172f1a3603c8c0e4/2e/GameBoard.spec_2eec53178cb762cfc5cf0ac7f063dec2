/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_GameBoard = require('./GameBoard'); /* istanbul ignore next */var _GameBoard2 = _interopRequireDefault(_GameBoard);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// GameBoard is responsible for render the game grid
// CONTRACTS
// 1. 

describe('GameBoard', function () {
  var props = /* istanbul ignore next */void 0,mountedGameBoard = /* istanbul ignore next */void 0;
  var gameBoard = function gameBoard() {
    if (!mountedGameBoard) {
      mountedGameBoard = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameBoard2.default, props));

    }
    return mountedGameBoard;
  };

  beforeEach(function () {
    props = {};
    mountedGameBoard = undefined;
  });

  it('render the GameBoard', function () {
    var gameBoardWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameBoard2.default, /* istanbul ignore next */null));
    expect(gameBoardWrapper).toMatchSnapshot();
  });

  it('render a div as root element', function () {
    var gameBoardElement = gameBoard().find('div');
    var wrappingDiv = gameBoardElement.first();
    expect(wrappingDiv.children()).toEqual(gameBoard().children());
  });
});