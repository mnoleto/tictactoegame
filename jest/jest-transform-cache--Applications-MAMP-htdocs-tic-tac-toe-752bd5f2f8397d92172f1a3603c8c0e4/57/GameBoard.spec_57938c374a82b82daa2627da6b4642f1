/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_GameBoard = require('./GameBoard'); /* istanbul ignore next */var _GameBoard2 = _interopRequireDefault(_GameBoard);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// GameBoard is responsible for render the game board
describe('<GameBoard/>', function () {
  var props = /* istanbul ignore next */void 0,mountedGameBoard = /* istanbul ignore next */void 0;
  var gameBoard = function gameBoard() {
    if (!mountedGameBoard) {
      mountedGameBoard = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameBoard2.default, props));

    }
    return mountedGameBoard;
  };

  beforeEach(function () {
    props = {
      board: ['E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E'],
      onClickButton: jest.fn() };

    mountedGameBoard = undefined;
  });

  it('should render the GameBoard', function () {
    var gameBoardWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameBoard2.default, props));
    expect(gameBoardWrapper).toMatchSnapshot();
  });

  it('should render a div as root element', function () {
    var gameBoardElement = gameBoard().find('div');
    var wrappingDiv = gameBoardElement.first();
    expect(wrappingDiv.hasClass('game-board')).toBeTruthy();
    expect(wrappingDiv.children()).toEqual(gameBoard().children());
  });

  it('should render 9 buttons', function () {
    var buttonElements = gameBoard().find('button');
    expect(buttonElements.length).toEqual(9);
  });
});