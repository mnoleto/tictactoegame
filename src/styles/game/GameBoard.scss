.game-board {
  display: block;
  margin: 0 auto 30px;
  padding: 0 10%;
  width: 100%;
  &:after {
    clear: both;
    content: "";
    display: block;
  }
}

@media (min-width: 768px) {
  .game-board {
    width: 740px;
  }
}

@media (min-width: 992px) {
  .game-board {
    width: 960px;
  }
}

@media (min-width: 1200px) {
  .game-board {
    width: 1170px;
  }
}