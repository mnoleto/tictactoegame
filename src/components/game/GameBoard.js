import React, { Component, PropTypes } from 'react';
import GameButton from './GameButton';
import '../../styles/game/GameBoard.scss';

class GameBoard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { board, onClickButton} = this.props;

    let buttons;
    if(board) {
      buttons = board.map((value, index) => {
        return(
          <GameButton
            key={'button_' + index}
            index={index}
            label={value}
            onClickButton={onClickButton} />
        );
      });
    }

    return (
      <div className="game-board">
        {buttons}
      </div>
    );
  }
}

GameBoard.propTypes = {
  board: PropTypes.array,
  onClickButton: PropTypes.func
}

export default GameBoard;