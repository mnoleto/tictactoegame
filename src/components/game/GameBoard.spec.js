import React from 'react';
import { mount, shallow } from 'enzyme';
import GameBoard from './GameBoard';

// GameBoard is responsible for render the game grid
describe('GameBoard', () => {
  let props, mountedGameBoard;
  const gameBoard = () => {
    if(!mountedGameBoard) {
      mountedGameBoard = mount(
        <GameBoard {...props} />
      );
    }
    return mountedGameBoard;
  };

  beforeEach(() => {
    props = {
      board: ['E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', ],
      onClickButton: jest.fn()
    };
    mountedGameBoard = undefined;
  });

  it('should render the GameBoard', () => {
    const gameBoardWrapper = shallow(<GameBoard {...props} />);
    expect(gameBoardWrapper).toMatchSnapshot();
  });

  it('should render a div as root element', () => {
    const gameBoardElement = gameBoard().find('div');
    let wrappingDiv = gameBoardElement.first();
    expect(wrappingDiv.children()).toEqual(gameBoard().children());
  });

  it('should render 9 buttons', () => {
    const buttonElements = gameBoard().find('button');
    expect(buttonElements.length).toEqual(9);
  })
});