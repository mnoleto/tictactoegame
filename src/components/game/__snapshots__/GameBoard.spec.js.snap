exports[`GameBoard should render the GameBoard 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="game-board">
    <GameButton
        index={0}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={1}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={2}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={3}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={4}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={5}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={6}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={7}
        label="E"
        onClickButton={[Function]} />
    <GameButton
        index={8}
        label="E"
        onClickButton={[Function]} />
</div>,
  "nodes": Array [
    <div
      className="game-board">
      <GameButton
            index={0}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={1}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={2}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={3}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={4}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={5}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={6}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={7}
            label="E"
            onClickButton={[Function]} />
      <GameButton
            index={8}
            label="E"
            onClickButton={[Function]} />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <GameBoard
        board={
                Array [
                        "E",
                        "E",
                        "E",
                        "E",
                        "E",
                        "E",
                        "E",
                        "E",
                        "E",
                      ]
        }
        onClickButton={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": GameBoard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "board": Array [
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
          ],
          "onClickButton": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="game-board">
          <GameButton
                    index={0}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={1}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={2}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={3}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={4}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={5}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={6}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={7}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={8}
                    label="E"
                    onClickButton={[Function]} />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="game-board">
          <GameButton
                    index={0}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={1}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={2}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={3}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={4}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={5}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={6}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={7}
                    label="E"
                    onClickButton={[Function]} />
          <GameButton
                    index={8}
                    label="E"
                    onClickButton={[Function]} />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GameBoard
    board={
        Array [
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
            "E",
          ]
    }
    onClickButton={[Function]} />,
}
`;
